WordPress/Classes/ViewRelated/Blog/QuickStartChecklistManager.swift

```diff
@@ -139,84 +92,32 @@ extension QuickStartChecklistManager: UITableViewDelegate {
-            return true
-        }
-
-        switch section {
-        case .todo:
-            return !(WPDeviceIdentification.isiPad() && !todoTours.isEmpty && indexPath.row == 0)
-        case .completed:
-            return true
-        }
-    }
-
-    func tours(at section: Int) -> [QuickStartTour] {
-        guard let section = Sections(rawValue: section) else {
-            return []
-        }
-
-        switch section {
-        case .todo:
-            return todoTours
-        case .completed:
-            return completedSectionCollapse ? completedTours : []
-        }
+        return !(WPDeviceIdentification.isiPad() && indexPath.row == 0)
     }
 
     func isCompleted(tour: QuickStartTour) -> Bool {
         return completedToursKeys.contains(tour.key)
     }
 
-    func tableView(_ tableView: UITableView, reloadCompletedSection collapsing: Bool) {
-        var indexPaths: [IndexPath] = []
-        for (index, _) in completedTours.enumerated() {
-            indexPaths.append(IndexPath(row: index, section: Sections.completed.rawValue))
-        }
-
-        tableView.perform(update: { tableView in
-            if collapsing {
-                tableView.insertRows(at: indexPaths, with: .fade)
-            } else {
-                tableView.deleteRows(at: indexPaths, with: .fade)
-            }
-        })
-
-        didTapHeader(collapsing)
-    }
-
     func tableView(_ tableView: UITableView, completeTourAt indexPath: IndexPath) {
-        let tour = todoTours[indexPath.row]
-        todoTours.remove(at: indexPath.row)
-        completedTours.append(tour)
+        let tour = tours[indexPath.row]
         completedToursKeys.insert(tour.key)
+        tableView.reloadRows(at: [indexPath], with: .fade)
 
         WPAnalytics.trackQuickStartStat(.quickStartListItemSkipped,
                                         properties: ["task_name": tour.analyticsKey],
                                         blog: blog)
+        QuickStartTourGuide.shared.complete(tour: tour, for: self.blog, postNotification: true)
+    }
+}
 
-        tableView.perform(update: { tableView in
-            tableView.deleteRows(at: [indexPath], with: .automatic)
-            let sections = IndexSet(integer: Sections.completed.rawValue)
-            tableView.reloadSections(sections, with: .fade)
-        }) { [weak self] tableView, _ in
-            DispatchQueue.main.async {
-                guard let self = self else {
-                    return
-                }
-                if self.shouldShowCompleteTasksScreen() {
-                    self.didTapHeader(self.completedSectionCollapse)
-                }
-                QuickStartTourGuide.shared.complete(tour: tour, for: self.blog, postNotification: false)
-                let sections = IndexSet(integer: Sections.todo.rawValue)
-                tableView.reloadSections(sections, with: .automatic)
-            }
-        }
+private extension QuickStartChecklistManager {
+    struct Constants {
```

Comment:
[nit] I like using enums for constants since enums are a pure namespace and can't be instantiated -- wdyt?
```suggestion
    enum Constants {
```

Reply:
Never thought of it this way 👀 
Great point actually, thanks for the tip!